src/validation.cpp

```diff
@@ -2680,7 +2682,8 @@ bool Chainstate::ConnectBlock(const CBlock& block, BlockValidationState& state,
                               tx_state.GetRejectReason(), tx_state.GetDebugMessage());
                 break;
             }
-            control.Add(std::move(vChecks));
+            if (control)
```

Comment:
In commit "validation: only create a CCheckQueueControl if it's actually going to be used" (bcde9153012b2ce3e9ff5b7cd1c6f9866453e727)

Seems like it would be good to change `parallel_script_checks ? &vChecks : nullptr` on line 2679 above to `control ? &vChecks : nullptr` to be checking for the `control` condition consistently in this function.

IMO, it would also be nice to have a comment here saying that if `control` is unset, checks were already performed in the `CheckInputScripts` call. Otherwise it looks like they are being skipped here.

Reply:
Mmm, this is kinda awkward either way. It'd be better imo to pass the actual `CCheckQueueControl&` as an argument to `CheckInputScripts`, but that makes this much more complicated for a bunch of reasons.

I pushed an additional commit to try to make this a little easier to read. Please let me know what you think. If it's over-complicating what's otherwise a pretty trivial PR, I can remove it.